src/main/scala/amailp/intellij/robot/psi/reference/LibraryToDefinitionReference.scala

```diff
@@ -0,0 +1,44 @@
+package amailp.intellij.robot.psi.reference
+
+import amailp.intellij.robot.psi.utils.ExtRobotPsiUtils
+import amailp.intellij.robot.psi.LibraryValue
+import com.intellij.codeInsight.lookup.{AutoCompletionPolicy, LookupElementBuilder}
+import com.intellij.psi.{PsiElement, PsiElementResolveResult, ResolveResult, PsiPolyVariantReferenceBase}
+import com.intellij.psi.util.QualifiedName
+import com.jetbrains.python.psi.PyFile
+import com.jetbrains.python.psi.stubs.{PyClassNameIndex, PyModuleNameIndex}
+import scala.collection.JavaConversions._
+import scala.collection.mutable
+
+
+class LibraryToDefinitionReference(library: LibraryValue) extends PsiPolyVariantReferenceBase[LibraryValue](library)
+  with ExtRobotPsiUtils {
+
+  override def getVariants = {
+    for {
+      lib <- currentRobotFile.getImportedLibraries
+    } yield LookupElementBuilder.create(lib.getText)
+      .withCaseSensitivity(false)
+      .withTypeText("Library", true)
+      .withAutoCompletionPolicy(AutoCompletionPolicy.GIVE_CHANCE_TO_OVERWRITE)
+      .asInstanceOf[AnyRef]
+  }.toArray
+
+  override def multiResolve(incompleteCode: Boolean): Array[ResolveResult] = {
+    val qName = QualifiedName.fromDottedString(library.getText)
+    var modules = mutable.MutableList[PyFile]()
```

Comment:
you do not need a mutable list here. looks like you can just use the return value of the **if** expression:

```scala
var module = if(...) {
  for {
    ...
  } *yield* psiFile
} else {
  PyModuleNameIndex.find(...)
}
```